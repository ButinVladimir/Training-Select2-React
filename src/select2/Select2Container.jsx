import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import './main.css';

const Select2Container = forwardRef(({ children }, ref) => (
  <div ref={ref} className="select2-container">
    {children}
  </div>
));

Select2Container.propTypes = {
  children: PropTypes.arrayOf(PropTypes.any).isRequired,
};

export default Select2Container;
