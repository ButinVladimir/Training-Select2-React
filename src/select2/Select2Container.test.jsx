import React from 'react';
import renderer from 'react-test-renderer';
import Select2Container from './Select2Container';

describe('Select2Container', () => {
  test('renders without crashing', () => {
    const component = renderer.create(
      <Select2Container ref={null}>
        <div>Child 1</div>
        <div>Child 2</div>
      </Select2Container>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
