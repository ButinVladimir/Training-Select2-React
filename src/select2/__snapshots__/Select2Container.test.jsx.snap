// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select2Container renders without crashing 1`] = `
<div
  className="select2-container"
>
  <div>
    Child 1
  </div>
  <div>
    Child 2
  </div>
</div>
`;
